### PR TITLE
(maint) Merge up 5.5.x to 6.4.x

### PR DESCRIPTION
**Pull request by mihaibuzgau**

### CODE REVIEW

File: ext/build_defaults.yaml [hunk 1]
```diff
@@ -16,3 +16,4 @@ apt_signing_server: 'apt.puppetlabs.com'
 apt_repo_url: 'http://apt.puppetlabs.com'
 apt_repo_path: '/opt/repository/incoming'
 tar_host: 'downloads.puppetlabs.com'
+nonfinal_gem_path: '/opt/repository-nightlies/downloads/gems/puppet5'
```
Comment (mwaggett) [inline ext/build_defaults.yaml]: This should probably be `puppet6-nightly` (and the 5.5.x branch should be `puppet5-nightly`)

File: lib/puppet/pops/puppet_stack.rb [hunk 2]
```diff
@@ -1,57 +1,59 @@
...
+    #
+    # ```rb
+    # Puppet::Pops::PuppetStack.top_of_stack
+    # ```
+    #
+    # To support testing, a given file that is an empty string, or nil
+    # as well as a nil line number are supported. Such stack frames
+    # will be represented with the text `unknown` and `0´ respectively.
+    module PuppetStack
+      @stack = Array.new
```
Comment (justinstoller) [inline lib/puppet/pops/puppet_stack.rb]: This looks correct, care will be needed when merging to master as this needs to be a threadlocal variable there.

File: ext/build_defaults.yaml (hunk 1 above)
Comment (melissa) [reply ext/build_defaults.yaml]: @mwaggett there is no `puppet5-nightly` at http://nightlies.puppet.com/downloads/gems/index_by_name.html. Should the 5.5.x branch still be changed?

**Comment by melissa** [comment]
I'm seeing weird failures in Travis
```
$ gem update --system && gem install bundler --no-document
Updating rubygems-update
Fetching: rubygems-update-3.1.1.gem (100%)
Successfully installed rubygems-update-3.1.1
Installing RubyGems 3.1.1
  Successfully built RubyGem
  Name: bundler
  Version: 2.1.0
  File: bundler-2.1.0.gem
bundler's executable "bundle" conflicts with /home/travis/.rvm/rubies/ruby-2.3.8/bin/bundle
Overwrite the executable? [yN]  
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#build-times-out-because-no-output-was-received
The build has been terminated
```

Do we know where the hosts we run Travis on come from? Has there been a change to them recently?

File: ext/build_defaults.yaml (hunk 1 above)
Comment (mwaggett) [reply ext/build_defaults.yaml]: @melissa I think they were originally shipped under `puppet5` and then removed so we could use the `-nightly` repo names. @gimmyxd should be able to confirm these paths.

**Comment by mihaibuzgau** [comment]
@melissa I'm re-doing the PR since the travis fix was merged

**Comment by mihaibuzgau** [comment]
this should be good to go

**Comment by melissa** [comment]
I'll merge once ci passes

File: ext/build_defaults.yaml (hunk 1 above)
Comment (gimmyxd) [reply ext/build_defaults.yaml]: puppet-agent pipeline is failing due to rubygems issues so the ship job never run. Also the job will still fail until the next release of packaging, so thats why there are no puppet5-nightly gems

**Pull request merged**